### DISCUSSION TITLE
Support package-vc-install to install magit

### DISCUSSION THREAD

**Discussion by kohnish** [magit/magit#5055] [Ideas]
It's pretty useful way for adding packages for users who just use pre-configured repositories.

**Comment by tarsius** [comment]
Tell me more! (Why does it not work now?)

**Reply by kohnish** [reply]
Sorry I was mistaken. It's already supported by specifying lisp dir.  
(package-vc-install '(magit :url "https://github.com/magit/magit.git" :lisp-dir "lisp"))

**Comment by kohnish** [comment]
Somehow automatic compilation done via package-vc-install ends up with a dysfunctional package even though installation seems to be successful.  
But it can be mitigated by adding make command in the top dir.
```
(let ((default-directory "~/.emacs.d/elpa/magit")
   (compile "make")))
```

**Reply by tarsius** [reply]
That's probably just yet another instance of the old loaded version of a package, leaking into the compilation result when installing another (later) version. `package.el` is completely broken in this regard.

Please verify this by uninstalling Magit and all dependencies, exiting and restarting Emacs, and only then (re-)installing.

**Reply by kohnish** [reply]
Indeed. that was the case, thank you very much!

**Reply by tarsius** [reply]
You're welcome!
